corehq/apps/events/models.py

```diff
@@ -57,6 +72,118 @@ def by_domain(cls, domain, most_recent_first=False):
             return cls.objects.filter(domain=domain).order_by('start_date')
         return cls.objects.filter(domain=domain)
 
+    @quickcache(['self.event_id'])
+    def get_expected_attendees(self):
+        """
+        Returns CommCareCase instances for the attendees for this Event.
+        """
+        # Attendee cases are associated with one or more Events using
+        # extension cases. The extension cases have case type
+        # EVENT_ATTENDEE_CASE_TYPE ('commcare-potential-attendee').
+        #
+        # The extension cases are owned by the Event's case-sharing
+        # group so that all mobile workers in the group get the attendee
+        # cases for the Event.
+        event_attendee_cases = self._get_ext_cases()
+
+        attendee_cases = []
+        for case in event_attendee_cases:
+            for index in case.indices:
+                if index.referenced_type == ATTENDEE_CASE_TYPE:
+                    attendee_cases.append(index.referenced_case)
+        return attendee_cases
+
+    def set_expected_attendees(self, attendee_cases):
+        """
+        Drops existing expected attendees, and creates extension cases
+        linking ``attendee_cases`` to this Event.
+
+        ``attendee_cases`` is a list of CommCareCase instances or case
+        IDs.
+        """
+        self.get_expected_attendees.clear(self)
+
+        self._close_ext_cases()
+
+        attendee_case_ids = (c if isinstance(c, str) else c.case_id
```

Comment:
I don't think it will ever be `CommCareCase` instances when we pass in the data directly from the frontend, but I think it also has to do with the expectation. When I read a method like `set_expected_attendees` I would expect to be able to provide a list of `CommCareCase` instances, since an attendee is a `CommCareCase` instance. There might exist a use-case in the future where we have a list of attendee cases, in which case we either strip out the id's beforehand or simply have this method do it automatically for us. Having this is a nice-to-have at the moment I think, but it keeps future code clean.